docs/bundles/SyliusResourceBundle/update_resource.rst

```diff
@@ -136,6 +136,29 @@ You can also perform more complex redirects, with parameters. For example:
                     route: app_genre_show
                     parameters: { id: $genreId }
 
+[API] Returning resource or no content
+--------------------------------------
+
+Depending on your app approach it can be useful to return a changed object or only 204 HTTP code, which indicates that everything worked smoothly.
+By default a 204 code is returned. If you would like to receive a whole object as a response you should set a `return_content` option to true.
```

Comment:
Ok, so we didn't standardize these two actions... It can be confusing that I can do a state machine transition using `updateAction` and I won't get object in a response but if I use `applyStateMachineTransitionAction` I will get it.

If we want to leave it that way we should clearly explain these two behaviors in the docs and also highlight that this option is only relevant for these two actions.